### PR TITLE
🥳[Feature] Add bech32 to enable ltc1 addresses

### LINKED ISSUE #14: 🥳 [Feature] Android Accepts and sends bech32 addresses
# Goal
Litewallet Android accepts and sends bech32 (1ltc...) addresses

# Definition of Done

- [ ]  Updated loafwallet-core submodule with bech32 changes
- [ ] Unit tests that checks different address types
- [ ] GIF in of the new action in PR

#  History 
https://github.com/litecoin-foundation/loafwallet-core/issues/22
https://github.com/litecoin-foundation/loafwallet-core/pull/19

### PR DESCRIPTION
**Pull request by kcw-grunt**
# We're back!

A few years ago, we tried to upgrade Litewallet for both platforms but the Bech32 upgrades were corrupting existing databases for Android users.  We stopped the rollout .

This PR is re-adding the changes and the code compiles but now we have to strategize the way to help users migrate.

Here are the tests matrix that need to be tested:

Scenario | Result Notes
----------|---------
New Install | Can the user send  L, M and 1ltc addresses?
Existing Install | Can the user send  L, M and 1ltc addresses?
Restore | Can the user send  L, M and 1ltc addresses?
New Install | Can the user receive  L, M and 1ltc addresses?
Existing Install | Can the user receive  L, M and 1ltc addresses?
Restore | Can the user receive  L, M and 1ltc addresses?
New Install | Can the user sync and get all  txs?
Existing Install | Can the user sync and get all  txs?
Restore | Can the user sync and get all  txs?

Closes #14

### CODE REVIEW

**Comment by kcw-grunt** [comment]
Update:
New Install on Google Pixel 4a:
- Received LTC to a L/M address
- Sent LTC to a 1ltc address

**Comment by kcw-grunt** [comment]
Testing:
New Install Restored privkey on Google Pixel 4a:
- [ ] Received LTC to a L/M address
- [ ] Sent LTC to a 1ltc address

**Comment by kcw-grunt** [comment]
@vsima .  Hey Vic! I'm stuck here. Maybe you can see the obvious problem.  What I'm trying to do is:
1. Display a fragment that appears when the Wallet is inited (BRAnimator, line 78).  This would ideally be right after the user unlocks.
2. Fragment (FragmentBalanceSeedReminder.kt)  has its layout (fragment_balance_seed_reminder.xml) and the user should be able to reveal the seed or dismiss.
3. Currently, the fragment never shows.
I'm really stuck here and frustrated 🤬.  Any clues?

Do you see anything obviously missing?

**Comment by vsima** [comment]
@kcw-grunt I pushed a commit to fix the UI issue & added logic to retrieve and display the seed phrase.

**Comment by kcw-grunt** [comment]
@vsima the button and modal looks good but somehow the database pulls a new seed instead of showing the existing on. For instance the first word for this seed is `extend` but the new one is `walnut`

There is some trigger starting creating a new seed there and we don't want that. Can you look at it.

|Recovery Seed Clip | Wrong new seed |
|-------------|---------------------------|
|1ltc-error-short|1ltc-error-wrong|

**Comment by kcw-grunt** [comment]
Ok we have a hybrid of @vsima work where I was able to not reset the seed call.
| Tested v2.8.3 (727)|
|-------------------|
|show-seed-phrase|

**Comment by kcw-grunt** [comment]
its looking stable 🤞🏾

**Comment by kcw-grunt** [comment]
still looking stable! 
@josikie @vsima can I get a look / review?

**Review by josikie — APPROVED** [review]
Wow! Cool!

**Comment by kcw-grunt** [comment]
Added the FP option
| Changing the FP|
|---------------|
|sync-preferences-clip|

The goal @vsima

File: app/src/main/jni/transition/PeerManager.c [hunk 1]
```diff
@@ -277,7 +278,7 @@ Java_com_breadwallet_wallet_BRPeerManager_create(JNIEnv *env, jobject thiz,
                             earliestKeyTime);
         _peerManager = BRPeerManagerNew(&BR_CHAIN_PARAMS, _wallet, (uint32_t) earliestKeyTime, _blocks,
                                         (size_t) blocksCount,
-                                        _peers, (size_t) peersCount, (double) fpRate);
+                                        _peers, (size_t) peersCount); //, t(double) fpRate
```
Comment (kcw-grunt) [inline app/src/main/jni/transition/PeerManager.c:281]: Hey @vsima  the new variable **(fpRate)** is what I was trying to add.  It looks like you commented it out.

Remember the goal is to inject fpRate (false positive rate) from the user preferences.  So, when they select it the **sync becomes much shorter**

File: app/src/main/jni/transition/PeerManager.c [hunk 2]
```diff
@@ -362,7 +363,7 @@ Java_com_breadwallet_wallet_BRPeerManager_putPeer(JNIEnv *env, jobject thiz, jby
 
 JNIEXPORT void JNICALL
 Java_com_breadwallet_wallet_BRPeerManager_createPeerArrayWithCount(JNIEnv *env, jobject thiz,
-                                                                   size_t peerCount) {
+                                                                   jint peerCount) {
```
Comment (kcw-grunt) [inline app/src/main/jni/transition/PeerManager.c:366]: Hey @vsima this is cruft....if I changed, it I made a mistake. I wasn't trying to touch peerCount at all

File: app/src/main/jni/transition/PeerManager.h [hunk 3]
```diff
@@ -40,20 +40,20 @@ Java_com_breadwallet_wallet_BRPeerManager_putPeer(JNIEnv *env, jobject thiz,
...
+Java_com_breadwallet_wallet_BRPeerManager_syncProgress(JNIEnv *env, jclass thiz,
                                                        int startHeight);
 
 JNIEXPORT jint JNICALL Java_com_breadwallet_wallet_BRPeerManager_getCurrentBlockHeight(JNIEnv *env,
-                                                                                       jobject thiz);
+                                                                                       jclass thiz);
 
 JNIEXPORT jint JNICALL Java_com_breadwallet_wallet_BRPeerManager_getEstimatedBlockHeight(
-        JNIEnv *env, jobject thiz);
+        JNIEnv *env, jclass thiz);
```
Comment (kcw-grunt) [inline app/src/main/jni/transition/PeerManager.h:56]: same here.

File: app/src/main/jni/transition/PeerManager.c (hunk 1 above)
Comment (vsima) [reply app/src/main/jni/transition/PeerManager.c]: Pulling the correct branch fixed the build issue. we can revert this commit.

**Comment by kcw-grunt** [comment]
Welcome @colin-yeoh  this our latest 🔥!

Please clone and take a look at the submodules for the project

**Pull request closed without merging**